src/ChargeDriftModels/InactiveLayerChargeDriftModel/InactiveLayerChargeDriftModel.jl

```diff
@@ -0,0 +1,109 @@
+"""
+    struct InactiveLayerChargeDriftModel{T <: SSDFloat} <: AbstractChargeDriftModel{T}
+        
+Charge drift model in which the electrons and holes drift along the electric field.
+There factors are considered in the mobility calculation: ionized impurities, neutral impurities, and acoustic phonon.
+
+Ref: [Dai _et al._ (2023)](https://doi.org/10.1016/j.apradiso.2022.110638)
+
+## Fields
+- `calculate_mobility::Function`: Mobility calculation function.
+- `neutral_imp_model::AbstractImpurityDensity{T}`: the neutral impurity density model. The default is a constant impurity density of 1e21 m⁻³.
+- `bulk_imp_model::AbstractImpurityDensity{T}`: the bulk impurity density model. The default is the defined (bulk) impurity density if the model is constructed in the config, otherwise the default is a constant impurity density of -1e16 m⁻³.
+- `surface_imp_model::AbstractImpurityDensity{T}`: the surface impurity density model. The default is the defined surface impurity density if the model is constructed in the config, otherwise the default is a constant impurity density of 0 m⁻³.
+
+## Extra field for constructing the model
+- `temperature::T`: temperature of the crystal (Kelvin).
+"""
+
+struct InactiveLayerChargeDriftModel{T <: SSDFloat} <: AbstractChargeDriftModel{T}
+    calculate_mobility::Function
+    temperature::T
+    neutral_imp_model::AbstractImpurityDensity{T}
+    bulk_imp_model::AbstractImpurityDensity{T}
+    surface_imp_model::AbstractImpurityDensity{T}
+end
+
+function InactiveLayerChargeDriftModel{T}(
+    temperature::T = T(90), # Kelvin
+    neutral_imp_model::AbstractImpurityDensity{T} = ConstantImpurityDensity{T}(T(1e21)),
+    bulk_imp_model::AbstractImpurityDensity{T} = ConstantImpurityDensity{T}(T(-1e16)),
+    surface_imp_model::AbstractImpurityDensity{T} = ConstantImpurityDensity{T}(T(0)),
+) where {T <: SSDFloat}
+
+    function calculate_mobility(pt::AbstractCoordinatePoint{T}, CC::Type{CC_type}) where {T <: SSDFloat, CC_type <: ChargeCarrier}
+        _calculate_mobility_with_impurities(
+            get_impurity_density(neutral_imp_model, pt),
+            get_impurity_density(bulk_imp_model, pt),
+            get_impurity_density(surface_imp_model, pt),
+            temperature, CC)
+    end
```

Comment:
Can you check whether the code runs faster/slower based on if this is a function which is part of the struct compared to if it's just a function in SSD (which could then look something like):

```suggestion
    function calculate_mobility(cdm::InactiveLayerChargeDriftModel{T}, pt::AbstractCoordinatePoint{T}, ::Type{CC}) where {T <: SSDFloat, CC <: ChargeCarrier}
        _calculate_mobility_with_impurities(
            get_impurity_density(cdm.neutral_imp_model, pt),
            get_impurity_density(cdm.bulk_imp_model, pt),
            get_impurity_density(cdm.surface_imp_model, pt),
            cdm.temperature, CC)
    end
```
?